### DISCUSSION TITLE
List of Fault Codes?

### DISCUSSION THREAD

**Discussion by muhkuh666** [nathanmarlor/foxess_modbus#672] [General]
Hi is there a listing of Faultcodes for H3 Inverters?

**Comment by canton7** [comment]
Yes, and the integration should be translating the fault codes into a descriptive text. What problem are you seeing?

**Reply by muhkuh666** [reply]
Thats good, thanks

**Comment by bartvz727** [comment]
![WhatsApp Image 2025-10-30 at 10 14 27 AM](https://github.com/user-attachments/assets/0db5eb0f-45fa-4db3-9b84-8e51b2572de6)
On this topic, I see faults on the actual screen on the front of my H3 (e.g. gridlostfault, gridfreqfault, etc.), but "none" in the sensor.h3_inverter_inverter_fault_code in HA.  Any thoughts on why this sensor isn't picking up the faults? For example, there are two grid faults at about 7:30am this morning in the inverter but not in the sensor.
<img width="571" height="563" alt="image" src="https://github.com/user-attachments/assets/46ec1adc-8fc2-4dc0-90c6-d4e873791736" />

**Reply by bartvz727** [reply]
It does seemed to have been working a few days ago, registering a grid lost fault, but not now?  Why would it change?  
<img width="596" height="437" alt="image" src="https://github.com/user-attachments/assets/eccba42f-b00e-44c5-9d42-9aed3515bf1b" />

**Reply by bartvz727** [reply]
Is it just because of the sampling rate? That is, if a fault happens and but is then cured in between data samples, it won't register in HA?  If so, what is that sampling rate?
